elastic-agent-client.proto

```diff
@@ -365,6 +365,10 @@ message CheckinObserved {
     // Supports provides information to the agent about extra features this client supports. Should always be included
     // on first checkin, and not again unless upon reconnect.
     repeated ConnectionSupports supports = 8;
+
+    // Optional field allowing the client to report its own PID. This is useful for applications like endpoint,
+    // where agent has less control over the binary.
+    uint64 checkin_pid = 9;
```

Comment:
```suggestion
    uint64 pid = 9;
```

The checkin is redundant I think, in some languages this will generate a getter like `checkin_observed().checkin_pid()`.